File: internal/core/domain/todo/todo.go
package todo

import (
	"time"

	"github.com/mecitsemerci/go-todo-app/internal/core/enum"
)

// Todo represents information
type Todo struct {
	ID          ID
	ProjectID   ID
	Title       string
	Description string
	Completed   bool
	Priority    enum.PriorityLevel
	CreatedAt   time.Time
	UpdatedAt   time.Time
}

// DefaultTodo is Default Todo
var DefaultTodo Todo

File: web/src/app/models/todo.ts
import { NgZone } from '@angular/core';

export interface Todo {
  id: string;
  title: string;
  description: string;
  priority_level: number;
  completed: boolean;
  create_at: string;
  update_at: string;
}

export interface CreateTodo {
  title: string;
  description?: string;
  priority_level?: number;
}

export interface CreateTodoResult {
  todo_id: string;
}

export interface UpdateTodo {
  title: string;
  description: string;
  priority_level: number;
  completed: boolean;
}


